mods/beds/api.lua

```diff
@@ -97,6 +99,20 @@ function beds.register_bed(name, def)
 			type = "fixed",
 			fixed = {0, 0, 0, 0, 0, 0},
 		},
+		on_destruct = function(pos)
```

Comment:
Adding it to only one part of the bad makes it very complicated (I will not say impossible, but not worth the complexity of the needed process) to make the second part break with the first one, we need a callback which can apply from both of the nodes, and here, there are two `on_destruct`, one per part.
Then, using a local variable might be faster, but more complicated. The only way I see is to create a local table, whose indexes are positions of dug part of beds, and values are always `1`, or something like `top` or `bottom`, depending on the destroyed part. In either way, it's more complicated than `minetest.get_meta(pos):[get/set]_int` which is an already implemented system to store datas related to a node, per node.